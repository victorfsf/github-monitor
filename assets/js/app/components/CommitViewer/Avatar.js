import React from 'react';
import PropTypes from 'prop-types';
import { GITHUB_AVATAR_URL } from 'constants/github';


const Avatar = (props) => {
  if (!props.id) {
    return (
      <div className="avatar mr-3 rounded justify-content-center d-flex">
        <i className=" fa fa-github-alt fa-2x align-middle" />
      </div>
    );
  }
  return (
    <img
      src={`${GITHUB_AVATAR_URL}${props.id}?v=4`}
      className="mr-3 rounded"
      alt={props.name}
      width="40px"
    />
  );
};


Avatar.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string.isRequired,
};


Avatar.defaultProps = {
  id: null,
};


export default Avatar;
