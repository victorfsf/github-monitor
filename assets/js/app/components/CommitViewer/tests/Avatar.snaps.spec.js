import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from 'app/components/CommitViewer/Avatar';


describe('Avatar', () => {
  let Component;
  let tree;

  test('does not render without name', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<Avatar />);
    expect(global.console.error).toHaveBeenCalledTimes(1);
  });

  test('renders without id', () => {
    Component = renderer.create(
      <Avatar name="Author" />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with id', () => {
    Component = renderer.create(
      <Avatar name="Author" id={1} />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
