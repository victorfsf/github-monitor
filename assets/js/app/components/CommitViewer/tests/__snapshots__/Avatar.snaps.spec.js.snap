// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders without id 1`] = `
<div
  className="avatar mr-3 rounded justify-content-center d-flex"
>
  <i
    className=" fa fa-github-alt fa-2x align-middle"
  />
</div>
`;

exports[`Avatar renders with id 1`] = `
<img
  alt="Author"
  className="mr-3 rounded"
  src="https://avatars1.githubusercontent.com/u/1?v=4"
  width="40px"
/>
`;
